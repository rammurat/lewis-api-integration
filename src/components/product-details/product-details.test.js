// @ts-nocheck
import React from 'react'
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux'
import Details from './product-details'
import toJson from 'enzyme-to-json'

import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });

const store = {
    subscribe: () => {},
    dispatch: () => {},
    getState: () => {}
}

test('Loading product snapshot', () => {
    const props = {
        productDetails: {
            title: 'product A',
            currencySign: 'Â£',
            priceNow: '20'
        },
        errMsg: '',
        isNoResult: false,
        isLoading: false,
    }
    const wrapper = shallow(<Provider store={store}><Details {...props} /></Provider>)
    expect(toJson(wrapper)).toMatchSnapshot()
});

test('Loading state snapshot', () => {
    const props = {
        productDetails: [],
        errMsg: '',
        isNoResult: false,
        isLoading: true
    }
    const wrapper = shallow(<Provider store={store}><Details {...props} /></Provider>)
    expect(toJson(wrapper)).toMatchSnapshot()
});

test('Error page snapshot', () => {
    const props = {
        productDetails: [],
        errMsg: 'There is an error from server',
        isNoResult: true,
        isLoading: false
    }
    const wrapper = shallow(<Provider store={store}><Details {...props} /></Provider>)
    expect(toJson(wrapper)).toMatchSnapshot()
});