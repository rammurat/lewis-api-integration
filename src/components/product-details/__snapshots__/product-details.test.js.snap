// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error page snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "subscribe": [Function],
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "onStateChange": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <Connect(ProductDetails)
    errMsg="There is an error from server"
    isLoading={false}
    isNoResult={true}
    productDetails={Array []}
  />
</ContextProvider>
`;

exports[`Loading product snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "subscribe": [Function],
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "onStateChange": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <Connect(ProductDetails)
    errMsg=""
    isLoading={false}
    isNoResult={false}
    productDetails={
      Object {
        "currencySign": "Â£",
        "priceNow": "20",
        "title": "product A",
      }
    }
  />
</ContextProvider>
`;

exports[`Loading state snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "subscribe": [Function],
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "onStateChange": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <Connect(ProductDetails)
    errMsg=""
    isLoading={true}
    isNoResult={false}
    productDetails={Array []}
  />
</ContextProvider>
`;
